src/s2geography/geoarrow_test.cc

```diff
@@ -202,3 +208,256 @@ TEST(GeoArrow, GeoArrowReaderReadWKTCollection) {
+  auto geog1 = reader.read_feature("POINT (0 1)");
+  auto geog2 = reader.read_feature("POINT (2 3)");
+
+  Writer writer;
+  nanoarrow::UniqueSchema schema;
+  nanoarrow::UniqueArray array;
+
+  InitSchemaGeoArrowPoint(schema.get());
+  writer.Init(schema.get());
+
+  writer.WriteGeography(*geog1);
+  writer.WriteGeography(*geog2);
+  writer.Finish(array.get());
+
+  EXPECT_EQ(array->length, 2);
+
+  // EXPECT_THAT(nanoarrow::ViewArrayAs<double_t>(array->children[0]), ElementsAre(DoubleEq(0.0), DoubleEq(2.0)));
+  auto xs = reinterpret_cast<const double*>(array->children[0]->buffers[1]);
+  EXPECT_DOUBLE_EQ(xs[0], 0.0);
+  EXPECT_DOUBLE_EQ(xs[1], 2.0);
+  // EXPECT_THAT(nanoarrow::ViewArrayAs<double_t>(array->children[1]), ElementsAre(DoubleEq(1.0), DoubleEq(3.0)));
+  auto ys = reinterpret_cast<const double*>(array->children[1]->buffers[1]);
+  EXPECT_DOUBLE_EQ(ys[0], 1.0);
+  EXPECT_DOUBLE_EQ(ys[1], 3.0);
+}
+
+TEST(GeoArrow, GeoArrowWriterPointProjected) {
+  s2geography::WKTReader reader;
+  auto geog1 = reader.read_feature("POINT (0 1)");
+  auto geog2 = reader.read_feature("POINT (2 3)");
+
+  Writer writer;
+  nanoarrow::UniqueSchema schema;
+  nanoarrow::UniqueArray array;
+
+  s2geography::geoarrow::ExportOptions options;
+  options.set_projection(s2geography::pseudo_mercator());
+  InitSchemaGeoArrowPoint(schema.get());
+  writer.Init(schema.get(), options);
+
+  writer.WriteGeography(*geog1);
+  writer.WriteGeography(*geog2);
+  writer.Finish(array.get());
+
+  EXPECT_EQ(array->length, 2);
+
+  auto xs = reinterpret_cast<const double*>(array->children[0]->buffers[1]);
+  EXPECT_DOUBLE_EQ(xs[0], 0.0);
+  EXPECT_DOUBLE_EQ(xs[1], 222638.98158654661);
+  auto ys = reinterpret_cast<const double*>(array->children[1]->buffers[1]);
+  EXPECT_DOUBLE_EQ(ys[0], 111325.14286638441);
+  EXPECT_DOUBLE_EQ(ys[1], 334111.17140195851);
+}
+
+TEST(GeoArrow, GeoArrowWriterPolylineTessellated) {
+  s2geography::WKTReader reader;
+  auto geog = reader.read_feature("LINESTRING (-64 45, 0 45)");
+
+  Writer writer;
+  nanoarrow::UniqueSchema schema;
+  GeoArrowSchemaInitExtension(schema.get(), GEOARROW_TYPE_LINESTRING);
+
+  nanoarrow::UniqueArray array;
+  writer.Init(schema.get());
+  writer.WriteGeography(*geog);
+  writer.Finish(array.get());
+
+  EXPECT_EQ(array->length, 1);
+  EXPECT_EQ(array->children[0]->length, 2);
+
+  // with tessellation -> more coordinates
+  nanoarrow::UniqueArray array2;
+  s2geography::geoarrow::ExportOptions options;
+  options.set_tessellate_tolerance(S1Angle::Radians(10000 / EARTH_RADIUS_METERS));
+  writer.Init(schema.get(), options);
+  writer.WriteGeography(*geog);
+  writer.Finish(array2.get());
+
+  EXPECT_EQ(array2->length, 1);
+  EXPECT_GT(array2->children[0]->length, 2);
+  EXPECT_EQ(array2->children[0]->length, 9);
+  // first coordinate is still the same
+  auto xs = reinterpret_cast<const double*>(array2->children[0]->children[0]->buffers[1]);
+  EXPECT_DOUBLE_EQ(xs[0], -64);
+  EXPECT_DOUBLE_EQ(xs[9], 0);
```

Comment:
I actually had a wrong index here (8 instead of 9, since the length is 9, the last index is 8 ;)) 
And it seems this was passing on non-windows because that random memory was zero-ed somehow?